src/persist-client/src/impl/metrics.rs

```diff
@@ -36,6 +36,8 @@ use crate::ShardId;
 #[derive(Debug)]
 pub struct Metrics {
     _vecs: MetricsVecs,
+    /// A copy of MetricsRegistry that can be cloned and passed to [mz_persist] metrics
+    pub registry: MetricsRegistry,
```

Comment:
> {"type":"test","name":"mz-persist::lib::postgres::tests::postgres_consensus","event":"failed","exec_time":0.244486964,"stdout":"thread 'postgres::tests::postgres_consensus' panicked at 'called `Result::unwrap()` on an `Err` value: AlreadyReg', /var/lib/buildkite-agent/builds/buildkite-builders-d43b1b5-i-01d8d995b711b0876-1/materialize/tests/src/ore/src/metrics.rs:223:54\nstack backtrace:\nnote: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.\n"}

A given metric name can only ever be registered once on a MetricsRegistry (including on its clones). That's the what this panic above from cargo test in CI is saying. I personally have found it pretty hard to reason about maintaining this invariant, except when we do all registration up front inside the single `mz_persist_client::metrics::#impl::Metrics::new` call that happens at process boot.

Now that I look at this in more detail, I think maybe we want something like the following?

- somewhere in mz_persist there is a PostgresConsensusMetrics with a `fn new(registry: &MetricsRegistry)` and a derived Clone
- inside the mz_persist_client Metrics::new call, we call this and store the resulting PostgresConsensusMetrics on Metrics
- ConsensusConfig::try_from then takes a `&PostgresConsensusMetrics` (yeah, awkward), which the postgres branch is free to clone

This has the downside that we'll create a PostgresConsensusMetrics (and thus export the postgres metrics) even when we're not using PostgresConsensus, but that's fine because the only other one we have is MemConsensus for tests.

It has the advantage that we could have two instances of `PostgresConsensus` in the same process (e.g. if different shards are pointed at different crdb instances). This was more relevant for aurora but it seems nice to keep around.

Thoughts?

Reply:
this was the first direction I took, but it made it very tricky to thread in a `ComputedGauge`, since the gauge needs access to the `pool` to define its collection function at registration time.

replacing the computed gauge with a normal one fixes this though! so going that route